dhis-2/dhis-api/src/main/java/org/hisp/dhis/calendar/impl/PersianCalendar.java

```diff
@@ -0,0 +1,683 @@
+    }
+
+    @Override
+    public int weeksInYear( int year )
+    {
+        DateTime dateTime = new DateTime( year, 1, 1, 0, 0,
+            ISOChronology.getInstance( DateTimeZone.getDefault() ) );
+        return dateTime.weekOfWeekyear().getMaximumValue();
+    }
+
+    @Override
+    public int isoWeek( DateTimeUnit dateTimeUnit )
+    {
+        DateTime dateTime = toIso( dateTimeUnit )
+            .toJodaDateTime( ISOChronology.getInstance( DateTimeZone.getDefault() ) );
+        return dateTime.getWeekOfWeekyear();
+    }
+
+    @Override
+    public int week( DateTimeUnit dateTimeUnit )
+    {
+        return isoWeek( dateTimeUnit );
+    }
+
+    @Override
+    public int isoWeekday( DateTimeUnit dateTimeUnit )
+    {
+
+        DateTime dateTime = toIso( dateTimeUnit )
+            .toJodaDateTime( ISOChronology.getInstance( DateTimeZone.getDefault() ) );
+        return dateTime.getDayOfWeek();
+    }
+
+    @Override
+    public int weekday( DateTimeUnit dateTimeUnit )
+    {
+
+        int dayOfWeek = (isoWeekday( dateTimeUnit ) + 1);
+
+        if ( dayOfWeek > 7 )
+        {
+            return 1;
+        }
+
+        return dayOfWeek;
+    }
+
+    @Override
+    public String nameOfMonth( int month )
+    {
+        if ( month > DEFAULT_I18N_MONTH_NAMES.length || month <= 0 )
+        {
+            return null;
+        }
+
+        return "persian." + DEFAULT_I18N_MONTH_NAMES[month - 1];
+    }
+
+    @Override
+    public String shortNameOfMonth( int month )
+    {
+        if ( month > DEFAULT_I18N_MONTH_SHORT_NAMES.length || month <= 0 )
+        {
+            return null;
+        }
+
+        return "persian." + DEFAULT_I18N_MONTH_SHORT_NAMES[month - 1];
+    }
+
+    @Override
+    public String nameOfDay( int day )
+    {
+        if ( day > DEFAULT_I18N_DAY_NAMES.length || day <= 0 )
+        {
+            return null;
+        }
+
+        return "persian." + DEFAULT_I18N_DAY_NAMES[day - 1];
+    }
+
+    @Override
+    public String shortNameOfDay( int day )
+    {
+        if ( day > DEFAULT_I18N_DAY_SHORT_NAMES.length || day <= 0 )
+        {
+            return null;
+        }
+
+        return "persian." + DEFAULT_I18N_DAY_SHORT_NAMES[day - 1];
+    }
+
+    private int getYearTotal( int year )
+    {
+
+        if ( CONVERSION_MAP.get( year ) == null )
+        {
+            throw new InvalidCalendarParamatersException(
+                "Illegal PERSIAN year, must be between " + START_PERSIAN.getYear() + " and " +
+                    STOP_PERSIAN.getYear() + " was given " + year );
+        }
+
+        if ( CONVERSION_MAP.get( year )[0] == 0 )
+        {
+            for ( int j = 1; j <= 12; j++ )
+            {
+                CONVERSION_MAP.get( year )[0] += CONVERSION_MAP.get( year )[j];
+            }
+        }
+
+        return CONVERSION_MAP.get( year )[0];
+    }
+
+    @Override
+    public DateTimeUnit minusYears( DateTimeUnit dateTimeUnit, int years )
```

Comment:
Make sure that public methods come first in the file, not a mix of public/private..